gateway/ln-gateway/src/actor.rs

```diff
@@ -111,10 +111,12 @@ impl GatewayActor {
 
     pub async fn stop_subscribing_htlcs(&mut self) -> Result<()> {
         if let Some(sender) = &self.sender {
-            sender
+            let _ = sender
                 .send(Arc::new(AtomicBool::new(true)))
                 .await
-                .expect("Couldn't send signal to shutdown HTLC thread");
+                .map_err(|e| {
+                    warn!("Couldn't send shutdown signal to HTLC thread: {:?}", e);
+                });
```

Comment:
Wondering if logging+forwarding the error is an anti-pattern. Either it will be swallowed further up, in which case the swallowing code should log, or it will lead to a panic further up, logging the error too.

Reply:
Ah yeah, got a little ahead of myself here. Here we should propagate the error up. Fixed.

The other `expect` that I changed I think should be swallowed, and I added a comment. It's running in the event loop thread, so returning `None` is enough to get the thread to exit.